src/main/java/dto/CategoryDTO.java

```diff
@@ -0,0 +1,41 @@
+package dto;
+
+import java.util.Objects;
+
+public class CategoryDTO {
+    private int id;
+    private String name;
+
+    public int getId(){
+        return id;
+    }
+    public void setId(int id){
+        this.id=id;
+    }
+    public String getName(){
+        return name;
+    }
+    public void setName(String name){
+        this.name=name;
+    }
+
+    public static CategoryDTO createCategory(int id, String name) {
+        CategoryDTO categoryDTO = new CategoryDTO();
+        categoryDTO.setId(id);
+        categoryDTO.setName(name);
+        return categoryDTO;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CategoryDTO that = (CategoryDTO) o;
+        return id == that.id && Objects.equals(name, that.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, name);
+    }
```

Comment:
Where do you use hashcode and equals?

Reply:
deleted the hashcode.
 Equals remained because we are not comparing links but objects